torch/csrc/jit/attributes.h

```diff
@@ -90,15 +92,35 @@ struct Attributes {
     }
   }
   bool hasAttribute(Symbol name) const {
+    JIT_ASSERT(name.is_attr());
     return find(name,false) != values_.end();
   }
+  // We want direct string accessors, as it is nicer to use than
+  // hasAttribute(Symbol::attr("blah"))
+  //
+  // For some reason, &Attributes<Node>::hasAttribute in pybind11 is able to
+  // give the pybind11 metaprogramming machinery "the right type", but
+  // the equivalent looking lambda [](Attributes<Node>& a, const std::string&)
+  // doesn't work!  So instead we define the methods on the class so we can
```

Comment:
What's the pybind error? It's not great that we're polluting the class with those methods.

Reply:
It's not a compile time error, but when I try to run it at runtime, pybind11 complains that there is no eligible dispatch for the types that are passed in. Here's one example:

```
======================================================================
ERROR: test_inplace_flags (__main__.TestJit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_jit.py", line 643, in test_inplace_flags
    self.assertTrue(op.hasAttribute('inplace'))
TypeError: hasAttribute(): incompatible function arguments. The following argument types are supported:
    1. (self: torch::jit::Attributes<torch::jit::Node>, arg0: str) -> bool

Invoked with: %2 : Double(1), %3 : Handle = ^RegularFn()(%0)
, 'inplace'

```

Reply:
I see. It would be nice if we could figure out a hack that we could put only in the Python bindings 😕

Reply:
I mean, what I would much rather know is why eta expansion doesn't work here. But I don't even know what this error message means, and I wanted to get the patch out...